### ISSUE TITLE
Broken release version 1.5.0

### ISSUE THREAD

**Issue by Lothiraldan** [chainer/chainer#8586]
TLDR: Pip 2.3.0 has been released with the new dependency solver (https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020) which can leads pip to test multiple versions of chainer which includes 1.5.0, which is a broken release, breaking the pip install process.

Chainer release 1.5.0 is broken and only show an error trying to install it:

```
$ pip install "chainer==1.5.0"
Collecting chainer==1.5.0
  Using cached chainer-1.5.0.tar.gz (144 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/lothiraldan/.virtualenvs/comet3.7/bin/python3.7 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-53n1uxb3/chainer/setup.py'"'"'; __file__='"'"'/tmp/pip-install-53n1uxb3/chainer/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-61dcnzte
         cwd: /tmp/pip-install-53n1uxb3/chainer/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-53n1uxb3/chainer/setup.py", line 7, in <module>
        import chainer_setup_build
    ModuleNotFoundError: No module named 'chainer_setup_build'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output. 
```

This shouldn't be a problem as the package is old and is likely not used by no-one anymore. But pip 20.3 has been released today and in some cases (when backtracking), it will try older version of the conflicting package. If this happens with chainer, here is what might happens:

```
INFO: pip is looking at multiple versions of chainer to determine which version is compatible with other requirements. This could take a while.
Collecting chainer
  Downloading chainer-7.4.0.tar.gz (1.0 MB)
Requirement already satisfied: setuptools in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (49.6.0)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-7.2.0.tar.gz (1.0 MB)
Requirement already satisfied: setuptools in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (49.6.0)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-7.1.0.tar.gz (1.0 MB)
Requirement already satisfied: setuptools in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (49.6.0)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-7.0.0.tar.gz (1.0 MB)

...

  Downloading chainer-1.6.1.tar.gz (907 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.6.0.tar.gz (904 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.5.1.tar.gz (864 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.5.0.3.tar.gz (864 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.5.0.2.tar.gz (191 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.5.0.1.tar.gz (147 kB)
Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.5/site-packages (from chainer->-r /src/test_requirements_py35.txt (line 5)) (1.15.0)
  Downloading chainer-1.5.0.tar.gz (144 kB)
[91m    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-emk8uoth/chainer_ff810dbcb6624941a09b91bec19f771c/setup.py'"'"'; __file__='"'"'/tmp/pip-install-emk8uoth/chainer_ff810dbcb6624941a09b91bec19f771c/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-miyh1sgu
         cwd: /tmp/pip-install-emk8uoth/chainer_ff810dbcb6624941a09b91bec19f771c/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-emk8uoth/chainer_ff810dbcb6624941a09b91bec19f771c/setup.py", line 7, in <module>
        import chainer_setup_build
    ImportError: No module named 'chainer_setup_build'
```

Basically all chainer versions are tried before trying version 1.5.0 and the broken release stop the version conflict resolution process.

I would recommend yanking the broken version in order to avoid users hitting the same issue, some documentation about yanking a release if you are not familiar with: https://pypi.org/help/#yanked

**Comment by kmaehashi** [comment]
@Lothiraldan Thanks for letting us know about it.
(1) How can I reproduce the error? If I understand correctly, there should be no backtracking as Chainer v1.5.0.1 and v1.5.0 has exactly the same requirements.
(2) If we yank chainer v1.5.0, what happens to users using `chainer==1.5.0`?

**Comment by Lothiraldan** [comment]
1. I'm not sure, when trying the new pip release, it horribly backtracked without giving me the offending packages for multiple hours. I think when backtracking it might try all packages in requirements.
2. I think that people explicitly asking for the version `1.5.0` will still get it, but it looks broken for all python versions or platform as the file `chainer_setup_build.py` is missing.

**Comment by kmaehashi** [comment]
Thanks! I've yanked v1.5.0 release.
